src/stan_math_backend/Expression_gen.ml

```diff
@@ -417,9 +417,23 @@ and pp_compiler_internal_fn ad ut f ppf es =
   | Some FnReadData -> read_data ut ppf es
   | Some FnReadParam -> (
     match es with
-    | {Expr.Fixed.pattern= Lit (Str, base_type); _} :: dims ->
-        pf ppf "@[<hov 2>in__.%s(@,%a)@]" base_type (list ~sep:comma pp_expr)
-          dims
+    | {Expr.Fixed.pattern= Lit (Str, constraint_string); meta= emeta}
+      :: {Expr.Fixed.pattern= Lit (Int, n_constraint_args_str); _} :: args ->
+        let n_constraint_args = int_of_string n_constraint_args_str in
+        let constraint_args, dims = List.split_n args n_constraint_args in
+        if String.is_empty constraint_string then
```

Comment:
I think I'd slightly favor "" (if it has to be a string), because 1) it's one less special constant value that we have to keep track of throughout the code, and 2) in all other cases that string represents part of the `read_constrain_` function name, so it makes some kind of sense to be empty string because it's not calling any constraint. I agree "identity" is more informative, but at this point it's generating backend-specific code that's packed into a pretty unreadable state anyway. This would also be solved by refactoring internal functions into variant types